pdr_backend/lake/table_bronze_pdr_slots.py

```diff
@@ -0,0 +1,203 @@
+                pl.col("timestamp_right").fill_null(pl.col("last_event_timestamp")),
+            ]
+        )
+        .drop(["trueval", "last_event_timestamp"])
+        .rename(
+            {
+                "trueval_right": "trueval",
+                "timestamp_right": "last_event_timestamp",
+            }
+        )
+        .select(initial_schema_keys)
+    )
+
+    # update dfs
+    dfs[bronze_pdr_slots_table_name] = slots_df
+    return dfs
+
+
+def _process_predictions(
+    dfs: Dict[str, pl.DataFrame], ppss: PPSS
+) -> Dict[str, pl.DataFrame]:
+    """
+    @description
+        Perform post-fetch processing on the data
+        1. Find predictions( within the update
+
+    """
+    # get predictions within the update
+    predictions_df, _ = pick_df_and_ids_on_period(
+        target=dfs["pdr_predictions"],
+        start_timestamp=ppss.lake_ss.st_timestamp,
+        finish_timestamp=ppss.lake_ss.fin_timestamp,
+    )
+
+    # get existing bronze_predictions we'll be updating
+    slots_df = dfs[bronze_pdr_slots_table_name]
+
+    # add columns that are going to be populated
+    slots_df = slots_df.with_columns(pair=pl.lit(None))
+    slots_df = slots_df.with_columns(source=pl.lit(None))
+    slots_df = slots_df.with_columns(timeframe=pl.lit(None))
+
+    initial_schema_keys = slots_df.schema.keys()
+
+    # do work to join from pdr_payout onto bronze_pdr_predictions
+    slots_df = (
+        slots_df.join(predictions_df, on=["contract"], how="left")
+        .with_columns(
+            [
+                pl.col("pair").fill_null(pl.col("pair")),
+                pl.col("source").fill_null(pl.col("source")),
+                pl.col("timeframe").fill_null(pl.col("timeframe")),
+            ]
+        )
+        .drop(["pair", "source", "timeframe"])
+        .rename(
+            {
+                "pair_right": "pair",
+                "source_right": "source",
+                "timeframe_right": "timeframe",
+            }
+        )
+        .select(initial_schema_keys)
+    )
+
+    slots_df = slots_df.unique(subset=["ID"])
+
+    # update dfs
+    dfs[bronze_pdr_slots_table_name] = slots_df
+    return dfs
+
+
+@enforce_types
+def get_bronze_pdr_slots_df(
+    gql_dfs: Dict[str, pl.DataFrame], ppss: PPSS
+) -> pl.DataFrame:
+    """
+    @description
+        Updates/Creates clean slots from existing raw tables
+    """
+
+    collision_ids = pl.Series([])
+    # retrieve pred ids that are already in the lake
+    if len(gql_dfs[bronze_pdr_slots_table_name]) > 0:
+        collision_ids = gql_dfs[bronze_pdr_slots_table_name].filter(
+            (pl.col("timestamp") >= ppss.lake_ss.st_timestamp)
+            & (pl.col("timestamp") <= ppss.lake_ss.fin_timestamp)
+        )["ID"]
+
+    # do post sync processing
+    gql_dfs = _process_slots(collision_ids, gql_dfs, ppss)
+    gql_dfs = _process_predictions(gql_dfs, ppss)
+    gql_dfs = _process_truevals(gql_dfs, ppss)
+
```

Comment:
[_process_payouts]
We need to process payouts, such that the slot is updated... Why? Because when the payout event is triggered, the `bronze_predictions` table (1) should already have been updated.

Also, since predictions were already submitted for this slot, our `predictions blob` should already have a bunch of users registered in there...

We should then find all of the users...
```
- prediction_id
- user_id
- timestamp
```

And join/update the blob such that their stake/payouts are also included...

We'll then get all users who were paid out... and join w/ the latest records from **bronze_predictions** that should be updated to include [predval, stake, payout, etc...]. We should then update the blob for the user to have the rest of the data we want, such that all the following data is saved per-user inside the slot.
```
- prediction_id
- user_id
- timestamp
- predval
- stake
- payout
```